array_api_compat/numpy/_info.py

```diff
@@ -326,7 +326,7 @@ def devices(self):
 
         Returns
         -------
-        devices : list of str
+        devices : list[Device]
```

Comment:
In numpy it is a string

Reply:
Strictly speaking, in numpy it is a `Literal["cpu"]`.
This is a fairly philosophical thing I doubt warrants much discussion.
My personal, very debatable opinion is that I'd rather stick to the abstract Device object rather than `str`.

The returned variable type in the two snippets below should be the same to static type checkers:

```python
import array_api_compat.numpy as xp
d = xp.__array_namespace_info__().devices()
```
vs.
```python
import numpy as np
from array_api_compat import array_namespace
xp = array_namespace(np.asarray(0))  # returns an abstracts array_api_types.Namespace
d = xp.__array_namespace_info__().devices()
```

(it doesn't today because `array-api-types` doesn't exist yet, but it will eventually).

Reply:
Numpy returns a string, not a `Device`, so why change this comment?

Reply:
because the user shouldn't think of it as a string, but as an opaque device object.